rasa/cli/test.py

```diff
@@ -83,23 +83,53 @@ def test_core(args: argparse.Namespace) -> None:
         )
 
     else:
-        test_compare(args.model, stories, output)
+        test_compare_core(args.model, stories, output)
 
 
 def test_nlu(args: argparse.Namespace) -> None:
     from rasa.test import test_nlu, test_nlu_with_cross_validation
+    from rasa.train import _get_valid_config
+    from rasa.nlu.utils import validate_pipeline_yaml
+    from rasa.nlu.model import InvalidModelError
+    import os
 
     nlu_data = get_validated_path(args.nlu, "nlu", DEFAULT_DATA_PATH)
     nlu_data = data.get_nlu_directory(nlu_data)
 
-    if not args.cross_validation:
+    if len(args.config) == 1:
+        args.config = args.config[0]
+        config_path = os.path.abspath(args.config)
+
+        if os.path.isdir(config_path):
+            files = os.listdir(args.config)
+            args.config = []
+            for file in files:
+                try:
+                    validate_pipeline_yaml(file)
+                except InvalidModelError:
+                    continue
+
+                valid_pipeline = _get_valid_config(file, ["pipeline", "language"])
+                args.config.append(valid_pipeline)
+
+    if isinstance(args.config, list):
+        logger.info("Multiple configs specified, running nlu comparison mode.")
+
```

Comment:
Can you move the check from line 108 to here? Otherwise you are only checking if the config files are valid, if the user provided a folder. Something like:

```
...

    if len(args.config) == 1:
        args.config = args.config[0]
        if os.path.isdir(os.path.abspath(args.config)):
            args.config = os.listdir(args.config)

    if isinstance(args.config, list):
        logger.info("Multiple configs specified, running nlu comparison mode.")

        config_files = []
        for file in args.config:
            try:
                validate_pipeline_yaml(file)
                config_files.append(file)
            except InvalidModelError:
                logger.debug("Ignoring file '{}' as it is not a valid config file.".format(file))
                continue

        output = args.report or "nlu_comparison_results"
        test_compare_nlu(
            configs=config_files,
            nlu=nlu_data,
            output=output,
            runs=args.runs,
            exclusion_percentages=args.percentages,
        )
...
```